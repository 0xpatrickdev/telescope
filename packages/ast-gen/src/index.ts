export * from './types';
export * from './amino';
export * from './client';
export * from './messages';
export * from './utils';