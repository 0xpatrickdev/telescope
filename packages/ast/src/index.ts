export * from './encoding';
export * from './clients';
export * from './registry';
export * from './services';
export * from './utils';
export * from './bundle';
export * from './types';