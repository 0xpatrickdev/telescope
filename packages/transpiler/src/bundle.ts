import { ProtoStore } from '@osmonauts/proto-parser';
import { recursiveModuleBundle, importNamespace } from '@osmonauts/ast';
import * as dotty from 'dotty';
import { TelescopeInput } from './index';
import { join, relative, dirname } from 'path';

// TODO move to store
export const getPackages = (store: ProtoStore) => {
    return store.getProtos().reduce((m, proto) => {
        if (proto.proto.package) {
            m[proto.proto.package] = m[proto.proto.package] || []
            m[proto.proto.package].push(proto);
        }
        return m;
    }, {});
}

export const getPackagesBundled = (store: ProtoStore) => {
    const objectified = {};
    const pkgs = getPackages(store);
    Object.keys(pkgs).forEach(key => {
        const files = pkgs[key];
        dotty.put(objectified, key, {
            pkg: key,
            files
        });
    });
    return objectified;

}

export const bundlePackages = (store: ProtoStore, input: TelescopeInput) => {
    const allPackages = getPackagesBundled(store);
    return Object.keys(allPackages).map(base => {
        const pkgs = allPackages[base];
        const bundleVariables = {};
        const bundleFile = join(input.protoDir, base, 'bundle.ts');
        const importPaths = [];
        parsePackage(pkgs, bundleFile, importPaths, bundleVariables);
        const body = recursiveModuleBundle(bundleVariables);
        return {
            bundleFile,
            importPaths,
            base,
            body
        };
    });
}

export const parsePackage = (obj, bundleFile, importPaths, bundleVariables) => {
    if (!obj) return;
    if (obj.pkg && obj.files) {
        obj.files.forEach(file => {
            createFileBundle(obj, file.filename, bundleFile, importPaths, bundleVariables);
            // TODO better namespacing for aminos/messages/encoding:
            // file.outFiles.forEach(outFile =>
            //     createFileBundle(obj, outFile.filename, bundleFile, importPaths, bundleVariables)
            // );
        });
        return;
    }
    Object.keys(obj).forEach(mini => {
        parsePackage(obj[mini], bundleFile, importPaths, bundleVariables);
    })
}


let counter = 0;
const createFileBundle = (obj, filename, bundleFile, importPaths, bundleVariables) => {
    let rel = relative(dirname(bundleFile), filename);
    if (!rel.startsWith('.')) rel = `./${rel}`;
    const variable = `_${counter++}`;
    importPaths.push(importNamespace(variable, rel));
    dotty.put(bundleVariables, obj.pkg + '.__export', true);
    dotty.put(bundleVariables, obj.pkg + '.' + variable, true);
}