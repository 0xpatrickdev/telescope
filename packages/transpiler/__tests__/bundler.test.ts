import { TelescopeBuilder } from '../src';
import { ProtoStore } from '@osmonauts/proto-parser'
import { getPackagesBundled } from '../src/bundle'

const store = new ProtoStore(__dirname + '/../../../__fixtures__/chain1');
store.traverseAll();

const telescope = new TelescopeBuilder({
    outPath: __dirname + '/../../../__fixtures__/output1',
    protoDir: __dirname + '/../../../__fixtures__/chain1'
});

it('bundle', () => {
    const bundle = getPackagesBundled(store);
    console.log(bundle);
});
