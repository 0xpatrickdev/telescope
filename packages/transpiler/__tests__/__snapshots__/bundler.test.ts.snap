// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundlePackages 1`] = `
Object {
  "cosmos": Object {
    "base": "cosmos",
    "bundle": "./chain1/cosmos/bundle.ts",
    "code": "import * as _2 from \\"../../../packages/transpiler/cosmos/app/v1alpha1/config\\";
import * as _3 from \\"../../../packages/transpiler/cosmos/app/v1alpha1/module\\";
import * as _4 from \\"../../../packages/transpiler/cosmos/app/v1alpha1/query\\";
import * as _5 from \\"../../../packages/transpiler/cosmos/auth/v1beta1/auth\\";
import * as _6 from \\"../../../packages/transpiler/cosmos/auth/v1beta1/genesis\\";
import * as _7 from \\"../../../packages/transpiler/cosmos/auth/v1beta1/query\\";
import * as _8 from \\"../../../packages/transpiler/cosmos/authz/v1beta1/authz\\";
import * as _9 from \\"../../../packages/transpiler/cosmos/authz/v1beta1/event\\";
import * as _10 from \\"../../../packages/transpiler/cosmos/authz/v1beta1/genesis\\";
import * as _11 from \\"../../../packages/transpiler/cosmos/authz/v1beta1/query\\";
import * as _12 from \\"../../../packages/transpiler/cosmos/authz/v1beta1/tx\\";
import * as _13 from \\"../../../packages/transpiler/cosmos/bank/v1beta1/authz\\";
import * as _14 from \\"../../../packages/transpiler/cosmos/bank/v1beta1/bank\\";
import * as _15 from \\"../../../packages/transpiler/cosmos/bank/v1beta1/genesis\\";
import * as _16 from \\"../../../packages/transpiler/cosmos/bank/v1beta1/query\\";
import * as _17 from \\"../../../packages/transpiler/cosmos/bank/v1beta1/tx\\";
import * as _18 from \\"../../../packages/transpiler/cosmos/base/abci/v1beta1/abci\\";
import * as _19 from \\"../../../packages/transpiler/cosmos/base/kv/v1beta1/kv\\";
import * as _20 from \\"../../../packages/transpiler/cosmos/base/query/v1beta1/pagination\\";
import * as _21 from \\"../../../packages/transpiler/cosmos/base/reflection/v1beta1/reflection\\";
import * as _22 from \\"../../../packages/transpiler/cosmos/base/reflection/v2alpha1/reflection\\";
import * as _23 from \\"../../../packages/transpiler/cosmos/base/snapshots/v1beta1/snapshot\\";
import * as _24 from \\"../../../packages/transpiler/cosmos/base/store/v1beta1/commit_info\\";
import * as _25 from \\"../../../packages/transpiler/cosmos/base/store/v1beta1/listening\\";
import * as _26 from \\"../../../packages/transpiler/cosmos/base/tendermint/v1beta1/query\\";
import * as _27 from \\"../../../packages/transpiler/cosmos/base/v1beta1/coin\\";
import * as _28 from \\"../../../packages/transpiler/cosmos/capability/v1beta1/capability\\";
import * as _29 from \\"../../../packages/transpiler/cosmos/capability/v1beta1/genesis\\";
import * as _30 from \\"../../../packages/transpiler/cosmos/crisis/v1beta1/genesis\\";
import * as _31 from \\"../../../packages/transpiler/cosmos/crisis/v1beta1/tx\\";
import * as _32 from \\"../../../packages/transpiler/cosmos/crypto/ed25519/keys\\";
import * as _33 from \\"../../../packages/transpiler/cosmos/crypto/hd/v1/hd\\";
import * as _34 from \\"../../../packages/transpiler/cosmos/crypto/keyring/v1/record\\";
import * as _35 from \\"../../../packages/transpiler/cosmos/crypto/multisig/keys\\";
import * as _36 from \\"../../../packages/transpiler/cosmos/crypto/secp256k1/keys\\";
import * as _37 from \\"../../../packages/transpiler/cosmos/crypto/secp256r1/keys\\";
import * as _38 from \\"../../../packages/transpiler/cosmos/distribution/v1beta1/distribution\\";
import * as _39 from \\"../../../packages/transpiler/cosmos/distribution/v1beta1/genesis\\";
import * as _40 from \\"../../../packages/transpiler/cosmos/distribution/v1beta1/query\\";
import * as _41 from \\"../../../packages/transpiler/cosmos/distribution/v1beta1/tx\\";
import * as _42 from \\"../../../packages/transpiler/cosmos/evidence/v1beta1/evidence\\";
import * as _43 from \\"../../../packages/transpiler/cosmos/evidence/v1beta1/genesis\\";
import * as _44 from \\"../../../packages/transpiler/cosmos/evidence/v1beta1/query\\";
import * as _45 from \\"../../../packages/transpiler/cosmos/evidence/v1beta1/tx\\";
import * as _46 from \\"../../../packages/transpiler/cosmos/feegrant/v1beta1/feegrant\\";
import * as _47 from \\"../../../packages/transpiler/cosmos/feegrant/v1beta1/genesis\\";
import * as _48 from \\"../../../packages/transpiler/cosmos/feegrant/v1beta1/query\\";
import * as _49 from \\"../../../packages/transpiler/cosmos/feegrant/v1beta1/tx\\";
import * as _50 from \\"../../../packages/transpiler/cosmos/genutil/v1beta1/genesis\\";
import * as _51 from \\"../../../packages/transpiler/cosmos/gov/v1/genesis\\";
import * as _52 from \\"../../../packages/transpiler/cosmos/gov/v1/gov\\";
import * as _53 from \\"../../../packages/transpiler/cosmos/gov/v1/query\\";
import * as _54 from \\"../../../packages/transpiler/cosmos/gov/v1/tx\\";
import * as _55 from \\"../../../packages/transpiler/cosmos/gov/v1beta1/genesis\\";
import * as _56 from \\"../../../packages/transpiler/cosmos/gov/v1beta1/gov\\";
import * as _57 from \\"../../../packages/transpiler/cosmos/gov/v1beta1/query\\";
import * as _58 from \\"../../../packages/transpiler/cosmos/gov/v1beta1/tx\\";
import * as _59 from \\"../../../packages/transpiler/cosmos/group/v1/events\\";
import * as _60 from \\"../../../packages/transpiler/cosmos/group/v1/genesis\\";
import * as _61 from \\"../../../packages/transpiler/cosmos/group/v1/query\\";
import * as _62 from \\"../../../packages/transpiler/cosmos/group/v1/tx\\";
import * as _63 from \\"../../../packages/transpiler/cosmos/group/v1/types\\";
import * as _64 from \\"../../../packages/transpiler/cosmos/mint/v1beta1/genesis\\";
import * as _65 from \\"../../../packages/transpiler/cosmos/mint/v1beta1/mint\\";
import * as _66 from \\"../../../packages/transpiler/cosmos/mint/v1beta1/query\\";
import * as _67 from \\"../../../packages/transpiler/cosmos/msg/v1/msg\\";
import * as _68 from \\"../../../packages/transpiler/cosmos/nft/v1beta1/event\\";
import * as _69 from \\"../../../packages/transpiler/cosmos/nft/v1beta1/genesis\\";
import * as _70 from \\"../../../packages/transpiler/cosmos/nft/v1beta1/nft\\";
import * as _71 from \\"../../../packages/transpiler/cosmos/nft/v1beta1/query\\";
import * as _72 from \\"../../../packages/transpiler/cosmos/nft/v1beta1/tx\\";
import * as _73 from \\"../../../packages/transpiler/cosmos/orm/v1/orm\\";
import * as _74 from \\"../../../packages/transpiler/cosmos/orm/v1alpha1/schema\\";
import * as _75 from \\"../../../packages/transpiler/cosmos/params/v1beta1/params\\";
import * as _76 from \\"../../../packages/transpiler/cosmos/params/v1beta1/query\\";
import * as _77 from \\"../../../packages/transpiler/cosmos/slashing/v1beta1/genesis\\";
import * as _78 from \\"../../../packages/transpiler/cosmos/slashing/v1beta1/query\\";
import * as _79 from \\"../../../packages/transpiler/cosmos/slashing/v1beta1/slashing\\";
import * as _80 from \\"../../../packages/transpiler/cosmos/slashing/v1beta1/tx\\";
import * as _81 from \\"../../../packages/transpiler/cosmos/staking/v1beta1/authz\\";
import * as _82 from \\"../../../packages/transpiler/cosmos/staking/v1beta1/genesis\\";
import * as _83 from \\"../../../packages/transpiler/cosmos/staking/v1beta1/query\\";
import * as _84 from \\"../../../packages/transpiler/cosmos/staking/v1beta1/staking\\";
import * as _85 from \\"../../../packages/transpiler/cosmos/staking/v1beta1/tx\\";
import * as _86 from \\"../../../packages/transpiler/cosmos/tx/signing/v1beta1/signing\\";
import * as _87 from \\"../../../packages/transpiler/cosmos/tx/v1beta1/service\\";
import * as _88 from \\"../../../packages/transpiler/cosmos/tx/v1beta1/tx\\";
import * as _89 from \\"../../../packages/transpiler/cosmos/upgrade/v1beta1/query\\";
import * as _90 from \\"../../../packages/transpiler/cosmos/upgrade/v1beta1/tx\\";
import * as _91 from \\"../../../packages/transpiler/cosmos/upgrade/v1beta1/upgrade\\";
import * as _92 from \\"../../../packages/transpiler/cosmos/vesting/v1beta1/tx\\";
import * as _93 from \\"../../../packages/transpiler/cosmos/vesting/v1beta1/vesting\\";
export namespace cosmos {
  export namespace app {
    export const v1alpha1 = { ..._2,
      ..._3,
      ..._4
    };
  }
  export namespace auth {
    export const v1beta1 = { ..._5,
      ..._6,
      ..._7
    };
  }
  export namespace authz {
    export const v1beta1 = { ..._8,
      ..._9,
      ..._10,
      ..._11,
      ..._12
    };
  }
  export namespace bank {
    export const v1beta1 = { ..._13,
      ..._14,
      ..._15,
      ..._16,
      ..._17
    };
  }
  export namespace base {
    export namespace abci {
      export const v1beta1 = { ..._18
      };
    }
    export namespace kv {
      export const v1beta1 = { ..._19
      };
    }
    export namespace query {
      export const v1beta1 = { ..._20
      };
    }
    export namespace reflection {
      export const v1beta1 = { ..._21
      };
      export const v2alpha1 = { ..._22
      };
    }
    export namespace snapshots {
      export const v1beta1 = { ..._23
      };
    }
    export namespace store {
      export const v1beta1 = { ..._24,
        ..._25
      };
    }
    export namespace tendermint {
      export const v1beta1 = { ..._26
      };
    }
    export const v1beta1 = { ..._27
    };
  }
  export namespace capability {
    export const v1beta1 = { ..._28,
      ..._29
    };
  }
  export namespace crisis {
    export const v1beta1 = { ..._30,
      ..._31
    };
  }
  export namespace crypto {
    export const ed25519 = { ..._32
    };
    export namespace hd {
      export const v1 = { ..._33
      };
    }
    export namespace keyring {
      export const v1 = { ..._34
      };
    }
    export const multisig = { ..._35
    };
    export const secp256k1 = { ..._36
    };
    export const secp256r1 = { ..._37
    };
  }
  export namespace distribution {
    export const v1beta1 = { ..._38,
      ..._39,
      ..._40,
      ..._41
    };
  }
  export namespace evidence {
    export const v1beta1 = { ..._42,
      ..._43,
      ..._44,
      ..._45
    };
  }
  export namespace feegrant {
    export const v1beta1 = { ..._46,
      ..._47,
      ..._48,
      ..._49
    };
  }
  export namespace genutil {
    export const v1beta1 = { ..._50
    };
  }
  export namespace gov {
    export const v1 = { ..._51,
      ..._52,
      ..._53,
      ..._54
    };
    export const v1beta1 = { ..._55,
      ..._56,
      ..._57,
      ..._58
    };
  }
  export namespace group {
    export const v1 = { ..._59,
      ..._60,
      ..._61,
      ..._62,
      ..._63
    };
  }
  export namespace mint {
    export const v1beta1 = { ..._64,
      ..._65,
      ..._66
    };
  }
  export namespace msg {
    export const v1 = { ..._67
    };
  }
  export namespace nft {
    export const v1beta1 = { ..._68,
      ..._69,
      ..._70,
      ..._71,
      ..._72
    };
  }
  export namespace orm {
    export const v1 = { ..._73
    };
    export const v1alpha1 = { ..._74
    };
  }
  export namespace params {
    export const v1beta1 = { ..._75,
      ..._76
    };
  }
  export namespace slashing {
    export const v1beta1 = { ..._77,
      ..._78,
      ..._79,
      ..._80
    };
  }
  export namespace staking {
    export const v1beta1 = { ..._81,
      ..._82,
      ..._83,
      ..._84,
      ..._85
    };
  }
  export namespace tx {
    export namespace signing {
      export const v1beta1 = { ..._86
      };
    }
    export const v1beta1 = { ..._87,
      ..._88
    };
  }
  export namespace upgrade {
    export const v1beta1 = { ..._89,
      ..._90,
      ..._91
    };
  }
  export namespace vesting {
    export const v1beta1 = { ..._92,
      ..._93
    };
  }
}",
  },
  "cosmos_proto": Object {
    "base": "cosmos_proto",
    "bundle": "./chain1/cosmos_proto/bundle.ts",
    "code": "import * as _1 from \\"../../../packages/transpiler/cosmos_proto/cosmos\\";
export const cosmos_proto = { ..._1
};",
  },
  "cosmwasm": Object {
    "base": "cosmwasm",
    "bundle": "./chain1/cosmwasm/bundle.ts",
    "code": "import * as _94 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/genesis\\";
import * as _95 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/ibc\\";
import * as _96 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/proposal\\";
import * as _97 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/query\\";
import * as _98 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/tx\\";
import * as _99 from \\"../../../packages/transpiler/cosmwasm/wasm/v1/types\\";
export namespace cosmwasm {
  export namespace wasm {
    export const v1 = { ..._94,
      ..._95,
      ..._96,
      ..._97,
      ..._98,
      ..._99
    };
  }
}",
  },
  "gogoproto": Object {
    "base": "gogoproto",
    "bundle": "./chain1/gogoproto/bundle.ts",
    "code": "import * as _100 from \\"../../../packages/transpiler/gogoproto/gogo\\";
export const gogoproto = { ..._100
};",
  },
  "google": Object {
    "base": "google",
    "bundle": "./chain1/google/bundle.ts",
    "code": "import * as _101 from \\"../../../packages/transpiler/google/api/annotations\\";
import * as _102 from \\"../../../packages/transpiler/google/api/auth\\";
import * as _103 from \\"../../../packages/transpiler/google/api/backend\\";
import * as _104 from \\"../../../packages/transpiler/google/api/billing\\";
import * as _105 from \\"../../../packages/transpiler/google/api/client\\";
import * as _106 from \\"../../../packages/transpiler/google/api/config_change\\";
import * as _107 from \\"../../../packages/transpiler/google/api/consumer\\";
import * as _108 from \\"../../../packages/transpiler/google/api/context\\";
import * as _109 from \\"../../../packages/transpiler/google/api/control\\";
import * as _110 from \\"../../../packages/transpiler/google/api/distribution\\";
import * as _111 from \\"../../../packages/transpiler/google/api/documentation\\";
import * as _112 from \\"../../../packages/transpiler/google/api/endpoint\\";
import * as _113 from \\"../../../packages/transpiler/google/api/error_reason\\";
import * as _114 from \\"../../../packages/transpiler/google/api/field_behavior\\";
import * as _115 from \\"../../../packages/transpiler/google/api/http\\";
import * as _116 from \\"../../../packages/transpiler/google/api/httpbody\\";
import * as _117 from \\"../../../packages/transpiler/google/api/label\\";
import * as _118 from \\"../../../packages/transpiler/google/api/launch_stage\\";
import * as _119 from \\"../../../packages/transpiler/google/api/log\\";
import * as _120 from \\"../../../packages/transpiler/google/api/logging\\";
import * as _121 from \\"../../../packages/transpiler/google/api/metric\\";
import * as _122 from \\"../../../packages/transpiler/google/api/monitored_resource\\";
import * as _123 from \\"../../../packages/transpiler/google/api/monitoring\\";
import * as _124 from \\"../../../packages/transpiler/google/api/quota\\";
import * as _125 from \\"../../../packages/transpiler/google/api/resource\\";
import * as _126 from \\"../../../packages/transpiler/google/api/routing\\";
import * as _127 from \\"../../../packages/transpiler/google/api/service\\";
import * as _128 from \\"../../../packages/transpiler/google/api/source_info\\";
import * as _129 from \\"../../../packages/transpiler/google/api/system_parameter\\";
import * as _130 from \\"../../../packages/transpiler/google/api/usage\\";
import * as _131 from \\"../../../packages/transpiler/google/api/visibility\\";
import * as _132 from \\"../../../packages/transpiler/google/logging/type/http_request\\";
import * as _133 from \\"../../../packages/transpiler/google/logging/type/log_severity\\";
import * as _134 from \\"../../../packages/transpiler/google/logging/v2/log_entry\\";
import * as _135 from \\"../../../packages/transpiler/google/logging/v2/logging_config\\";
import * as _136 from \\"../../../packages/transpiler/google/logging/v2/logging_metrics\\";
import * as _137 from \\"../../../packages/transpiler/google/logging/v2/logging\\";
import * as _138 from \\"../../../packages/transpiler/google/longrunning/operations\\";
import * as _139 from \\"../../../packages/transpiler/google/protobuf/any\\";
import * as _140 from \\"../../../packages/transpiler/google/protobuf/api\\";
import * as _141 from \\"../../../packages/transpiler/google/protobuf/descriptor\\";
import * as _142 from \\"../../../packages/transpiler/google/protobuf/duration\\";
import * as _143 from \\"../../../packages/transpiler/google/protobuf/empty\\";
import * as _144 from \\"../../../packages/transpiler/google/protobuf/field_mask\\";
import * as _145 from \\"../../../packages/transpiler/google/protobuf/source_context\\";
import * as _146 from \\"../../../packages/transpiler/google/protobuf/struct\\";
import * as _147 from \\"../../../packages/transpiler/google/protobuf/timestamp\\";
import * as _148 from \\"../../../packages/transpiler/google/protobuf/type\\";
import * as _149 from \\"../../../packages/transpiler/google/protobuf/wrappers\\";
import * as _150 from \\"../../../packages/transpiler/google/rpc/code\\";
import * as _151 from \\"../../../packages/transpiler/google/rpc/error_details\\";
import * as _152 from \\"../../../packages/transpiler/google/rpc/status\\";
export namespace google {
  export const api = { ..._101,
    ..._102,
    ..._103,
    ..._104,
    ..._105,
    ..._106,
    ..._107,
    ..._108,
    ..._109,
    ..._110,
    ..._111,
    ..._112,
    ..._113,
    ..._114,
    ..._115,
    ..._116,
    ..._117,
    ..._118,
    ..._119,
    ..._120,
    ..._121,
    ..._122,
    ..._123,
    ..._124,
    ..._125,
    ..._126,
    ..._127,
    ..._128,
    ..._129,
    ..._130,
    ..._131
  };
  export namespace logging {
    export const type = { ..._132,
      ..._133
    };
    export const v2 = { ..._134,
      ..._135,
      ..._136,
      ..._137
    };
  }
  export const longrunning = { ..._138
  };
  export const protobuf = { ..._139,
    ..._140,
    ..._141,
    ..._142,
    ..._143,
    ..._144,
    ..._145,
    ..._146,
    ..._147,
    ..._148,
    ..._149
  };
  export const rpc = { ..._150,
    ..._151,
    ..._152
  };
}",
  },
  "ibc": Object {
    "base": "ibc",
    "bundle": "./chain1/ibc/bundle.ts",
    "code": "import * as _153 from \\"../../../packages/transpiler/ibc/applications/transfer/v1/genesis\\";
import * as _154 from \\"../../../packages/transpiler/ibc/applications/transfer/v1/query\\";
import * as _155 from \\"../../../packages/transpiler/ibc/applications/transfer/v1/transfer\\";
import * as _156 from \\"../../../packages/transpiler/ibc/applications/transfer/v1/tx\\";
import * as _157 from \\"../../../packages/transpiler/ibc/applications/transfer/v2/packet\\";
import * as _158 from \\"../../../packages/transpiler/ibc/core/channel/v1/channel\\";
import * as _159 from \\"../../../packages/transpiler/ibc/core/channel/v1/genesis\\";
import * as _160 from \\"../../../packages/transpiler/ibc/core/channel/v1/query\\";
import * as _161 from \\"../../../packages/transpiler/ibc/core/channel/v1/tx\\";
import * as _162 from \\"../../../packages/transpiler/ibc/core/client/v1/client\\";
import * as _163 from \\"../../../packages/transpiler/ibc/core/client/v1/genesis\\";
import * as _164 from \\"../../../packages/transpiler/ibc/core/client/v1/query\\";
import * as _165 from \\"../../../packages/transpiler/ibc/core/client/v1/tx\\";
import * as _166 from \\"../../../packages/transpiler/ibc/core/commitment/v1/commitment\\";
import * as _167 from \\"../../../packages/transpiler/ibc/core/connection/v1/connection\\";
import * as _168 from \\"../../../packages/transpiler/ibc/core/connection/v1/genesis\\";
import * as _169 from \\"../../../packages/transpiler/ibc/core/connection/v1/query\\";
import * as _170 from \\"../../../packages/transpiler/ibc/core/connection/v1/tx\\";
import * as _171 from \\"../../../packages/transpiler/ibc/core/port/v1/query\\";
import * as _172 from \\"../../../packages/transpiler/ibc/core/types/v1/genesis\\";
import * as _173 from \\"../../../packages/transpiler/ibc/lightclients/localhost/v1/localhost\\";
import * as _174 from \\"../../../packages/transpiler/ibc/lightclients/solomachine/v1/solomachine\\";
import * as _175 from \\"../../../packages/transpiler/ibc/lightclients/solomachine/v2/solomachine\\";
import * as _176 from \\"../../../packages/transpiler/ibc/lightclients/tendermint/v1/tendermint\\";
export namespace ibc {
  export namespace applications {
    export namespace transfer {
      export const v1 = { ..._153,
        ..._154,
        ..._155,
        ..._156
      };
      export const v2 = { ..._157
      };
    }
  }
  export namespace core {
    export namespace channel {
      export const v1 = { ..._158,
        ..._159,
        ..._160,
        ..._161
      };
    }
    export namespace client {
      export const v1 = { ..._162,
        ..._163,
        ..._164,
        ..._165
      };
    }
    export namespace commitment {
      export const v1 = { ..._166
      };
    }
    export namespace connection {
      export const v1 = { ..._167,
        ..._168,
        ..._169,
        ..._170
      };
    }
    export namespace port {
      export const v1 = { ..._171
      };
    }
    export namespace types {
      export const v1 = { ..._172
      };
    }
  }
  export namespace lightclients {
    export namespace localhost {
      export const v1 = { ..._173
      };
    }
    export namespace solomachine {
      export const v1 = { ..._174
      };
      export const v2 = { ..._175
      };
    }
    export namespace tendermint {
      export const v1 = { ..._176
      };
    }
  }
}",
  },
  "ics23": Object {
    "base": "ics23",
    "bundle": "./chain1/ics23/bundle.ts",
    "code": "import * as _0 from \\"../../../packages/transpiler/confio/proofs\\";
export const ics23 = { ..._0
};",
  },
  "osmosis": Object {
    "base": "osmosis",
    "bundle": "./chain1/osmosis/bundle.ts",
    "code": "import * as _177 from \\"../../../packages/transpiler/osmosis/claim/v1beta1/claim\\";
import * as _178 from \\"../../../packages/transpiler/osmosis/claim/v1beta1/genesis\\";
import * as _179 from \\"../../../packages/transpiler/osmosis/claim/v1beta1/params\\";
import * as _180 from \\"../../../packages/transpiler/osmosis/claim/v1beta1/query\\";
import * as _181 from \\"../../../packages/transpiler/osmosis/epochs/genesis\\";
import * as _182 from \\"../../../packages/transpiler/osmosis/epochs/query\\";
import * as _183 from \\"../../../packages/transpiler/osmosis/gamm/pool-models/balancer/balancerPool\\";
import * as _184 from \\"../../../packages/transpiler/osmosis/gamm/pool-models/balancer/tx\\";
import * as _185 from \\"../../../packages/transpiler/osmosis/gamm/v1beta1/genesis\\";
import * as _186 from \\"../../../packages/transpiler/osmosis/gamm/v1beta1/query\\";
import * as _187 from \\"../../../packages/transpiler/osmosis/gamm/v1beta1/tx\\";
import * as _188 from \\"../../../packages/transpiler/osmosis/incentives/gauge\\";
import * as _189 from \\"../../../packages/transpiler/osmosis/incentives/genesis\\";
import * as _190 from \\"../../../packages/transpiler/osmosis/incentives/params\\";
import * as _191 from \\"../../../packages/transpiler/osmosis/incentives/query\\";
import * as _192 from \\"../../../packages/transpiler/osmosis/incentives/tx\\";
import * as _193 from \\"../../../packages/transpiler/osmosis/lockup/genesis\\";
import * as _194 from \\"../../../packages/transpiler/osmosis/lockup/lock\\";
import * as _195 from \\"../../../packages/transpiler/osmosis/lockup/query\\";
import * as _196 from \\"../../../packages/transpiler/osmosis/lockup/tx\\";
import * as _197 from \\"../../../packages/transpiler/osmosis/mint/v1beta1/genesis\\";
import * as _198 from \\"../../../packages/transpiler/osmosis/mint/v1beta1/mint\\";
import * as _199 from \\"../../../packages/transpiler/osmosis/mint/v1beta1/query\\";
import * as _200 from \\"../../../packages/transpiler/osmosis/pool-incentives/v1beta1/genesis\\";
import * as _201 from \\"../../../packages/transpiler/osmosis/pool-incentives/v1beta1/gov\\";
import * as _202 from \\"../../../packages/transpiler/osmosis/pool-incentives/v1beta1/incentives\\";
import * as _203 from \\"../../../packages/transpiler/osmosis/pool-incentives/v1beta1/query\\";
import * as _204 from \\"../../../packages/transpiler/osmosis/store/v1beta1/tree\\";
import * as _205 from \\"../../../packages/transpiler/osmosis/superfluid/genesis\\";
import * as _206 from \\"../../../packages/transpiler/osmosis/superfluid/params\\";
import * as _207 from \\"../../../packages/transpiler/osmosis/superfluid/query\\";
import * as _208 from \\"../../../packages/transpiler/osmosis/superfluid/superfluid\\";
import * as _209 from \\"../../../packages/transpiler/osmosis/superfluid/tx\\";
import * as _210 from \\"../../../packages/transpiler/osmosis/txfees/v1beta1/feetoken\\";
import * as _211 from \\"../../../packages/transpiler/osmosis/txfees/v1beta1/genesis\\";
import * as _212 from \\"../../../packages/transpiler/osmosis/txfees/v1beta1/gov\\";
import * as _213 from \\"../../../packages/transpiler/osmosis/txfees/v1beta1/query\\";
export namespace osmosis {
  export namespace claim {
    export const v1beta1 = { ..._177,
      ..._178,
      ..._179,
      ..._180
    };
  }
  export namespace epochs {
    export const v1beta1 = { ..._181,
      ..._182
    };
  }
  export namespace gamm {
    export const v1beta1 = { ..._183,
      ..._184,
      ..._185,
      ..._186,
      ..._187
    };
  }
  export const incentives = { ..._188,
    ..._189,
    ..._190,
    ..._191,
    ..._192
  };
  export const lockup = { ..._193,
    ..._194,
    ..._195,
    ..._196
  };
  export namespace mint {
    export const v1beta1 = { ..._197,
      ..._198,
      ..._199
    };
  }
  export namespace poolincentives {
    export const v1beta1 = { ..._200,
      ..._201,
      ..._202,
      ..._203
    };
  }
  export namespace store {
    export const v1beta1 = { ..._204
    };
  }
  export const superfluid = { ..._205,
    ..._206,
    ..._207,
    ..._208,
    ..._209
  };
  export namespace txfees {
    export const v1beta1 = { ..._210,
      ..._211,
      ..._212,
      ..._213
    };
  }
}",
  },
  "secret": Object {
    "base": "secret",
    "bundle": "./chain1/secret/bundle.ts",
    "code": "import * as _214 from \\"../../../packages/transpiler/secret/compute/v1beta1/genesis\\";
import * as _215 from \\"../../../packages/transpiler/secret/compute/v1beta1/msg\\";
import * as _216 from \\"../../../packages/transpiler/secret/compute/v1beta1/query\\";
import * as _217 from \\"../../../packages/transpiler/secret/compute/v1beta1/types\\";
import * as _218 from \\"../../../packages/transpiler/secret/registration/v1beta1/genesis\\";
import * as _219 from \\"../../../packages/transpiler/secret/registration/v1beta1/msg\\";
import * as _220 from \\"../../../packages/transpiler/secret/registration/v1beta1/query\\";
import * as _221 from \\"../../../packages/transpiler/secret/registration/v1beta1/types\\";
import * as _222 from \\"../../../packages/transpiler/secret/registration/v1beta1/remote_attestation/types\\";
export namespace secret {
  export namespace compute {
    export const v1beta1 = { ..._214,
      ..._215,
      ..._216,
      ..._217
    };
  }
  export namespace registration {
    export const v1beta1 = { ..._218,
      ..._219,
      ..._220,
      ..._221
    };
    export namespace remote_attestation {
      export const v1beta1 = { ..._222
      };
    }
  }
}",
  },
  "tendermint": Object {
    "base": "tendermint",
    "bundle": "./chain1/tendermint/bundle.ts",
    "code": "import * as _223 from \\"../../../packages/transpiler/tendermint/abci/types\\";
import * as _224 from \\"../../../packages/transpiler/tendermint/crypto/keys\\";
import * as _225 from \\"../../../packages/transpiler/tendermint/crypto/proof\\";
import * as _226 from \\"../../../packages/transpiler/tendermint/libs/bits/types\\";
import * as _227 from \\"../../../packages/transpiler/tendermint/p2p/types\\";
import * as _228 from \\"../../../packages/transpiler/tendermint/types/block\\";
import * as _229 from \\"../../../packages/transpiler/tendermint/types/evidence\\";
import * as _230 from \\"../../../packages/transpiler/tendermint/types/params\\";
import * as _231 from \\"../../../packages/transpiler/tendermint/types/types\\";
import * as _232 from \\"../../../packages/transpiler/tendermint/types/validator\\";
import * as _233 from \\"../../../packages/transpiler/tendermint/version/types\\";
export namespace tendermint {
  export const abci = { ..._223
  };
  export const crypto = { ..._224,
    ..._225
  };
  export namespace libs {
    export const bits = { ..._226
    };
  }
  export const p2p = { ..._227
  };
  export const types = { ..._228,
    ..._229,
    ..._230,
    ..._231,
    ..._232
  };
  export const version = { ..._233
  };
}",
  },
}
`;
